### PR TITLE
Support to configure the networkMode and podMetadata of vineyard components.

### PR DESCRIPTION
**Pull request by dashanji**
### Ⅰ. Describe what this PR does

Part of https://github.com/fluid-cloudnative/fluid/pull/3749

### CODE REVIEW

File: pkg/ddc/vineyard/transform.go [hunk 1]
```diff
@@ -262,3 +285,54 @@ func (e *VineyardEngine) transformTieredStore(runtime *datav1alpha1.VineyardRunt
...
+	expectedMasterPortNum, expectedWorkerPortNum := 0, 0
+	if datav1alpha1.IsHostNetwork(runtime.Spec.Master.NetworkMode) {
+		expectedMasterPortNum = 2
+	}
+	if datav1alpha1.IsHostNetwork(runtime.Spec.Worker.NetworkMode) {
+		expectedWorkerPortNum = 2
+	}
+	expectedPortNum := expectedMasterPortNum + expectedWorkerPortNum
+
+	allocator, err := portallocator.GetRuntimePortAllocator()
```
Comment (cheyang) [inline pkg/ddc/vineyard/transform.go:299]: May I know where do you use `setupRuntimeAllocator` to init?

**Review by cheyang — CHANGES_REQUESTED** [review]

File: pkg/ddc/vineyard/transform.go (hunk 1 above)
Comment (cheyang) [reply pkg/ddc/vineyard/transform.go:299]: I think you should have the following code:

```golang
Cmd.Flags().StringVar(&portAllocatePolicy, "port-allocate-policy", "random", "Set port allocating policy, available choice is bitmap or random(default random).")

err = portallocator.SetupRuntimePortAllocator(mgr.GetClient(), pr, portAllocatePolicy, juicefs.GetReservedPorts)
	if err != nil {
		setupLog.Error(err, "failed to setup runtime port allocator")
		os.Exit(1)
	}
```

Comment (dashanji) [reply pkg/ddc/vineyard/transform.go:299]: Sorry, it's my fault. I have not noticed the init function of the `setupRuntimeAllocator`. Thanks for pointing it out.

**Review by cheyang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**